### DISCUSSION TITLE
Validation: Setting default decimal mark when UNA is not present

### DISCUSSION THREAD

**Discussion by Kaiawaia** [xlate/staedi#405] [Q&A]
Hi,

I'm using base="decimal" for some element types in my validation schema. Now I have a problem when the UNA segment is not present in an EDIFACT file. As I've seen in the source code the DFLT_DECIMAL_MARK constant is used in this case, which is a point (.) The problem is that in german locale I need the comma instead of the point for the decimal mark. Is there any possibility to change the default decimal mark for this case?

thanks and regards
Kai

**Comment by MikeEdgar** [comment]
Hi @Kaiawaia , thanks for starting this discussion. After reviewing the EDIFACT control specifications again, I think this is a bug. The documents for both v1-v3 [1] and v4 [2] (section 10 in both) say that either the comma or point may be used. Versions 1-3 allow for the UNA to limit to only comma or only a point, but version 4 does not.

Would you mind to open a bug report [issue](https://github.com/xlate/staedi/issues) for this and we can get it in to the next release of staedi.

[1] https://service.gefeg.com/jswg/v3/data/v1-9735.pdf
[2] https://service.gefeg.com/jswg/v4/data/v4-9735-1.pdf

**Reply by Kaiawaia** [reply]
Thanks for your fast reply, @MikeEdgar .

I've created this [issue](https://github.com/xlate/staedi/issues/406) for the problem.
